### ISSUE TITLE
`doResolveEditor` has fragile check on `override`

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#127211]
When I look into the `doResolveEditor` implementation there is one place where we resolve / populate the editor ID if not found in override options:

https://github.com/microsoft/vscode/blob/ee290519377dbe3e4deb3b325269297b3950030d/src/vs/workbench/services/editor/browser/editorService.ts#L587-L587

Everything is fine for passing in untyped editors, but when passing in typed editors I think we might run into a problem:

https://github.com/microsoft/vscode/blob/ee290519377dbe3e4deb3b325269297b3950030d/src/vs/workbench/services/editor/browser/editorService.ts#L576-L576

This line essentially takes the `options` as they are passed in if they are present and drops the `untypedEditor.options` from the `toUntyped` call. But `toUntyped` will set the `override` in most cases which means:
* if I open a typed editor with options, the editor will get overridden unless I provide `override` options
* if I open a typed editor and do not provide options, the editor will not get overridden

Isn't this strange?

**Comment by bpasero** [comment]
This seems to be causing https://github.com/microsoft/vscode/issues/127215

**Comment by lramos15** [comment]
Should I close this and #127215 since #127287 was merged?

**Comment by bpasero** [comment]
@lramos15 this issue here seems addressed but I can still reproduce https://github.com/microsoft/vscode/issues/127215 when changing this line:

https://github.com/microsoft/vscode/blob/2c63b5dc9aa171491529cd3a7cc07e457d4ed1d8/src/vs/workbench/services/editor/browser/editorService.ts#L1244-L1244

to this:

```ts
const editorPane = await this.openEditor(editor, undefined, groupId);
```

Which essentially means, anyone opening a untitled editor without disabling the override potentially triggers a new untitled input to be created?

**Comment by lramos15** [comment]
Closing because of #127287. I resolved #127215.
